core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -71,9 +73,19 @@ enum TransactionType {
   private TableMetadata base;
   private TableMetadata current;
   private boolean hasLastOpCommitted;
+  private final MetricsReporter reporter;
 
   BaseTransaction(
       String tableName, TableOperations ops, TransactionType type, TableMetadata start) {
+    this(tableName, ops, type, start, LoggingMetricsReporter.instance());
+  }
+
+  BaseTransaction(
```

Comment:
Because the transaction constructor is package-private, we can remove the old one, right? Or do we want to keep it for testing?

Reply:
the old constructor is still being used at a few places, so we either adjust those places to always pass a `LoggingMetricsReporter.instance()` or we leave the old constructor and pass `LoggingMetricsReporter.instance()` from there. I chose the latter approach as it requires less changes